include/cxxopts.hpp

```diff
@@ -2111,36 +2169,34 @@ OptionAdder::operator()
   std::string arg_help
 )
 {
-  std::string short_sw, long_sw;
-  std::tie(short_sw, long_sw) = values::parser_tool::SplitSwitchDef(opts);
-
-  if (!short_sw.length() && !long_sw.length())
-  {
-    throw_or_mimic<exceptions::invalid_option_format>(opts);
+  OptionNames all_option_names = values::parser_tool::split_option_names(opts);
+    // Note: All names will be non-empty!
+  std::string short_sw {""};
+  OptionNames long_option_names = all_option_names;
+  auto first_length_1_name_it = std::find_if(long_option_names.cbegin(), long_option_names.cend(),
+    [&](const std::string& name) { return name.length() == 1; }
+  );
+  if (first_length_1_name_it != long_option_names.cend()) {
+    auto have_second_length_1_name = std::any_of(first_length_1_name_it + 1, long_option_names.cend(),
+      [&](const std::string& name) { return name.length() == 1; }
+    );
+    if (have_second_length_1_name) {
+      throw_or_mimic<exceptions::invalid_option_format>(opts);
+    }
+    short_sw = *first_length_1_name_it;
+    long_option_names.erase(first_length_1_name_it);
```

Comment:
This seems a bit complicated. Are you basically trying to find one short name and then leave multiple long names?
I wonder if a simple loop would just be clearer. Or maybe you could use `remove_if` and then check if the remaining list is greater than 1.

Reply:
Well remove_if doesn't actually remove it, it only moves everything to the end, so I think this is what you want.

Reply:
Oh I just read that the values at the end have unspecified value, so they might get destroyed. In that case partition is probably the way to go.

Reply:
Wait, actually, no, that's not quite what happens. If I `std::remove_if()` on a vector, the moved strings at the end don't keep their original values. So, that won't work.